test/regress/expect/tpch0001_d/q10.txt

```diff
@@ -30,45 +30,61 @@ group by
-    -> PhysicOrder  (inccost=15919.98, cost=413.98, rows=90, memory=21780) (actual rows=3)
+    -> PhysicOrder  (inccost=11794.98, cost=413.98, rows=90, memory=21780) (actual rows=20)
         Output: c_custkey[0],c_name[1],{sum(l_extendedprice*1-l_discount)}[2],c_acctbal[3],n_name[4],c_address[5],c_phone[6],c_comment[7]
         Order by: {sum(l_extendedprice*1-l_discount)}[2]
-        -> PhysicHashAgg  (inccost=15506, cost=270, rows=90, memory=43560) (actual rows=3)
+        -> PhysicHashAgg  (inccost=11381, cost=270, rows=90, memory=43560) (actual rows=43)
             Output: {c_custkey}[0],{c_name}[1],{sum(l_extendedprice*1-l_discount)}[7],{c_acctbal}[2],{n_name}[4],{c_address}[5],{c_phone}[3],{c_comment}[6]
             Aggregates: sum(l_extendedprice[8]*1-l_discount[11])
             Group by: c_custkey[0], c_name[1], c_acctbal[2], c_phone[5], n_name[3], c_address[4], c_comment[6]
-            -> PhysicGather 1 : 10 (inccost=15236, cost=9, rows=90) (actual rows=8)
-                Output: c_custkey[0],c_name[1],c_acctbal[2],n_name[8],c_address[3],c_phone[4],c_comment[5],{l_extendedprice*1-l_discount}[9],l_extendedprice[10],{1-l_discount}[11],1,l_discount[12]
-                -> PhysicHashJoin  (inccost=15227, cost=1890, rows=90, memory=65100) (actual rows=0)
-                    Output: c_custkey[0],c_name[1],c_acctbal[2],n_name[8],c_address[3],c_phone[4],c_comment[5],{l_extendedprice*1-l_discount}[9],l_extendedprice[10],{1-l_discount}[11],1,l_discount[12]
-                    Filter: c_custkey[0]=o_custkey[13] and c_nationkey[7]=n_nationkey[14]
-                    -> PhysicRedistribute  (inccost=165, cost=15, rows=150) (actual rows=0)
-                        Output: c_custkey[0],c_name[1],c_acctbal[2],c_address[3],c_phone[4],c_comment[5],{1}[6],c_nationkey[7]
-                        -> PhysicScanTable customer (inccost=150, cost=150, rows=150) (actual rows=0)
-                            Output: c_custkey[0],c_name[1],c_acctbal[5],c_address[2],c_phone[4],c_comment[7],1,c_nationkey[3]
-                    -> PhysicRedistribute  (inccost=13172, cost=150, rows=1500) (actual rows=0)
-                        Output: n_name[0],{l_extendedprice*1-l_discount}[1],l_extendedprice[2],{1-l_discount}[3],l_discount[4],o_custkey[5],n_nationkey[6]
-                        -> PhysicNLJoin  (inccost=13022, cost=2450, rows=1500) (actual rows=0)
-                            Output: n_name[0],{l_extendedprice*1-l_discount}[2],l_extendedprice[3],{1-l_discount}[4],l_discount[5],o_custkey[6],n_nationkey[1]
-                            -> PhysicScanTable nation (inccost=25, cost=25, rows=25) (actual rows=0)
-                                Output: n_name[1],n_nationkey[0]
-                            -> PhysicRedistribute  (inccost=10547, cost=6, rows=60) (actual rows=0)
+            -> PhysicGather 1 : 10 (inccost=11111, cost=900, rows=90) (actual rows=140)
```

Comment:
why we have a Gather node here? It is essentially serialize the plan.

Reply:
currently I made hashagg only able to deal with singleton, therefore gather will be enforced below hashagg.
I tried enabling distribution for hashagg and gather can be place on top of hashagg, but it will result in inconsistent results for average expression.